pkg/providers/amifamily/suite_test.go

```diff
@@ -132,12 +132,43 @@ var _ = AfterSuite(func() {
 	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
 })
 
+type MockVersionProvider struct {
+	version string
+}
+
+func (m *MockVersionProvider) Get(ctx context.Context) string {
+	return m.version
+}
+
+func mockAmiProvider(version string) *amifamily.DefaultProvider {
```

Comment:
nit: it's not the AMI provider that's mocked, it's it's the version provider provided to it right? I'd do something along the lines of `amiProviderWithEKSVersionOverride(version string)`, but let me know what you think.